api/renter.go

```diff
@@ -8,25 +11,57 @@ import (
 
 	"github.com/NebulousLabs/Sia/build"
 	"github.com/NebulousLabs/Sia/modules"
+	"github.com/NebulousLabs/Sia/modules/renter"
 	"github.com/NebulousLabs/Sia/types"
 
 	"github.com/julienschmidt/httprouter"
 )
 
 var (
-	// TODO: Replace this function by accepting user input.
-	recommendedHosts = func() uint64 {
-		if build.Release == "dev" {
-			return 2
-		}
-		if build.Release == "standard" {
-			return 24
-		}
-		if build.Release == "testing" {
-			return 1
-		}
-		panic("unrecognized release constant in api")
-	}()
+	// recommendedHosts is the number of hosts that the renter will form
+	// contracts with if the value is not specified explicity in the call to
+	// SetSettings.
+	recommendedHosts = build.Select(build.Var{
+		Standard: uint64(30),
+		Dev:      uint64(2),
+		Testing:  uint64(1),
+	}).(uint64)
+
+	// requiredHosts specifies the minimum number of hosts that must be set in
+	// the renter settings for the renter settings to be valid. This minimum is
+	// there to prevent users from shooting themselves in the foot.
+	requiredHosts = build.Select(build.Var{
+		Standard: uint64(24),
+		Dev:      uint64(1),
+		Testing:  uint64(1),
+	}).(uint64)
+
+	// requiredParityPieces specifies the minimum number of parity pieces that
+	// must be used when uploading a file. This minimum exists to prevent users
+	// from shooting themselves in the foot.
+	requiredParityPieces = build.Select(build.Var{
+		Standard: int(12),
+		Dev:      int(0),
+		Testing:  int(0),
+	}).(int)
+
+	// requiredRedundancy specifies the minimum redundancy that will be
+	// accepted by the renter when uploading a file. This minimum exists to
+	// prevent users from shooting themselves in the foot.
+	requiredRedundancy = build.Select(build.Var{
+		Standard: float64(2.5),
+		Dev:      float64(1),
+		Testing:  float64(1),
+	}).(float64)
+
+	// requiredRenewWindow establishes the minimum allowed renew window for the
+	// renter settings. This minimum is here to prevent users from shooting
+	// themselves in the foot.
+	requiredRenewWindow = build.Select(build.Var{
+		Standard: types.BlockHeight(288),
+		Dev:      types.BlockHeight(1),
+		Testing:  types.BlockHeight(1),
+	}).(types.BlockHeight)
```

Comment:
I added API support for choosing redundancy and host count. Along with this API support, I added some constants to specify what the default values would be, and what the minimum allowed values would be. Minimum values are obviously easy to circumvent by recompiling, but if you are advanced enough to do that then you deserve whatever footguns you create for yourself.